src/components/auth/Login.jsx

```diff
@@ -1,15 +1,15 @@
 import { useEffect, useState } from 'react';
-
+import { useNavigate, useLocation } from 'react-router-dom';
 import LoginForm from './LoginForm';
 import { login } from '../../util/DataBaseRequests';
 import { useAuth } from '../../AuthProvider';
-import { useNavigate } from 'react-router-dom';
 
 const Login = () => {
   const [formData, setFormData] = useState({ email: '', password: '' });
   const [formErrors, setFormErrors] = useState({});
   const { isLoggedIn, setUserSession } = useAuth();
   const navigate = useNavigate();
+  const location = useLocation();
 
   const handleChange = (e) => {
```

Comment:
This function has a lot going on know. It could benefit from separating some concerns. Here we have form validation and login. Form validation can be abstracted to its own function.